app/services/support_interface/change_application_choice_course_option.rb

```diff
@@ -0,0 +1,37 @@
+module SupportInterface
+  class ChangeApplicationChoiceCourseOption
+    attr_reader :application_choice, :provider_id, :course_code, :study_mode, :site_code, :audit_comment
+
+    def initialize(application_choice_id:,
+                   provider_id:,
+                   course_code:,
+                   study_mode:,
+                   site_code:,
+                   audit_comment:)
+      @application_choice = ApplicationChoice.find(application_choice_id)
+      @provider_id = provider_id
+      @course_code = course_code
+      @site_code = site_code
+      @study_mode = study_mode
+      @audit_comment = audit_comment
+    end
+
+    def call
+      application_choice.update_course_option_and_associated_fields!(course_option,
```

Comment:
do we want to only restrict this to applications in `awaiting_provider_decisions` to avoid any mistakes? Also I was wondering about applications in the interview state, maybe we allow this service to deal with that as well, before we plug it into the support tool?

Reply:
Good idea. Amended to only allow applications in awaiting_provider_decision and interviewing states as the support requests contain applications in both of these staes.